### PR TITLE
Adding test suites for anomaly detection (smoke, one-shot, pattern, edge)

### PR DESCRIPTION
**Pull request by architjain19**
## Summary
This adds a few test suites covering all critical testing categories for the `roboqa-temporal` package - `anomaly-detection` feature. The test suite includes the following tests:

> 1. **Smoke Tests** (`tests/test_smoke.py`)
> Verify basic package functionality and imports

> 2. **One-Shot Tests** (`tests/test_one_shot.py`)
> Test single execution paths with minimal setup

> 3. **Edge Tests** (`tests/test_edge.py`)
> Test boundary conditions and unusual inputs

> 4. **Pattern Tests** (`tests/test_pattern.py`)
> Test common usage workflows and realistic scenarios

## Running Tests (Usage)

```bash
# To run all tests, use-
python3 -m pytest tests/ -v

# To run specific test categories, use-
python3 -m pytest tests/test_smoke.py -v
```

### CODE REVIEW

**Review by Dharineesh-Somisetty — COMMENTED** [review]
Thanks for these test cases, I will do similar thing with my feature too. Great work overall!!

**Comment by architjain19** [comment]
Merging the PR

File: tests/test_edge.py [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+    assert np.isinf(frame.points[0, 2])
+
+
+def test_detector_with_all_detectors_disabled():
+    """
+    author: architjain
+    reviewer: dharinesh
+    category: edge test
+    """
+    detector = AnomalyDetector(
```
Comment (IanQS) [inline tests/test_edge.py]: @architjain19 @Dharineesh-Somisetty 

Design question (I honestly have no idea) - if all the detectors are off, wouldn't throwing an error be appropriate?

**Review by IanQS — COMMENTED** [review]
Some design questions to be answered

File: tests/test_edge.py [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    assert anomaly.severity == 0.0
+
+
+def test_anomaly_with_max_severity():
+    """
+    author: architjain
+    reviewer: dharinesh
+    category: edge test
+    """
+    anomaly = Anomaly(
```
[the remark this answers is not in the document]
Comment (architjain19) [reply tests/test_edge.py:162]: Thanks! Severity is now explicitly bounded in one place: added `MIN_SEVERITY`/`MAX_SEVERITY` to `constants.py` and `Anomaly` now validates severity stays within [0.0, 1.0] on construction. This makes the 0.0–1.0 range the contract and catches out-of-range values early.


> We acknowledge the feedback and is a good point to clarify assert message.
> We tackle it with this implementation:
> ```python
> for col in ["temporal_score", "anomaly_score", "multimodal_health_score"]:
>    assert np.all(np.isfinite(df[col])), f"{col} contains non-finite values"
>    assert np.all(df[col] >= 0.0), f"{col} has values < 0.0"
>    assert np.all(df[col] <= 1.0), f"{col} has values > 1.0"
>  ```

File: tests/test_edge.py (hunk 1 above)
Comment (architjain19) [reply tests/test_edge.py:62]: Good catch, thanks! Having all detectors off is now treated as a misconfiguration. `AnomalyDetector` raises a clear `ValueError` if no detectors are enabled.

> We tackle this with:
> ```python
>        if not self.detectors:
>            raise ValueError(
>                "No detectors enabled. Enable at least one anomaly detector or adjust configuration."
>            )
>
>        # checking edge case here-
>        with pytest.raises(ValueError, match="No detectors enabled"):
>            AnomalyDetector(
>                enable_density_detection=False,
>                enable_spatial_detection=False,
>                enable_ghost_detection=False,
>                enable_temporal_detection=False,
>            )
>```
